src/docdb/nodes.ts

```diff
@@ -0,0 +1,151 @@
+		return await new Promise<any[]>((resolve, reject) => {
+		databases.toArray( (err, dbs: Array<Object>) => err ? reject(err) : resolve(dbs) );
+		});
+	}
+
+	async getDocDBDatabaseNodes(client, DocDBServerNodeInstance): Promise<INode[]> {
+		let databases = [];
+		databases = await this.listDatabases(client);
+		return databases.map(database => new DocDBDatabaseNode(database.id, DocDBServerNodeInstance));
+	}
+
+}
+
+export class DocDBDatabaseNode implements INode {
+	readonly contextValue: string = 'DocDbDatabase';
+
+	constructor(readonly id: string, readonly server: DocDBServerNode) {
+	}
+
+	get label(): string {
+		return this.id;
+	}
+
+	get iconPath(): any {
+		return {
+			light: path.join(__filename, '..', '..', '..', '..', 'resources', 'icons', 'light', 'Database.svg'),
+			dark: path.join(__filename, '..', '..', '..', '..', 'resources', 'icons', 'dark', 'Database.svg')
+		};
+	}
+
+	readonly collapsibleState = vscode.TreeItemCollapsibleState.Collapsed;
+
+	getDbLink(): string {
+		return 'dbs/' + this.id;
+	}
+
+	getChildren(): Promise<INode[]> {
+		return this.getCollections();
+	}
+
+	async getCollections(): Promise<INode[]> {
+		let dbLink: string = this.getDbLink();
+		let collections;
+		let parentNode = this;
+		let client = new DocumentClient(this.server.getEndpoint(), {masterKey: this.server.getPrimaryMasterKey() });
+		collections = await this.listCollections(dbLink, client);
+		return collections.map(collection => new DocDBCollectionNode(collection.id, parentNode));
+	}
+
+	async listCollections(databaseLink, client): Promise<any> {
+		let collections = await client.readCollections(databaseLink);
+		return await new Promise<any[]>((resolve, reject) => {
+			collections.toArray( (err, cols: Array<Object>) => err ? reject(err) : resolve(cols) );
+		});
+	}
+
+}
+
+
+export class DocDBCollectionNode implements INode {
+
+	constructor(readonly id: string, readonly db: DocDBDatabaseNode) {
+	}
+
+	readonly contextValue: string = 'DocDbCollection';
+
+	get label(): string {
+		return this.id;
+	}
+
+	get iconPath(): any {
+		return {
+			light: path.join(__filename, '..', '..', '..', '..', 'resources', 'icons', 'light', 'Collection.svg'),
+			dark: path.join(__filename, '..', '..', '..', '..', 'resources', 'icons', 'dark', 'Collection.svg'),
+		};
+	}
+	readonly collapsibleState = vscode.TreeItemCollapsibleState.None;
+
+	readonly command: Command = {
+		command: 'cosmosDB.openDocDBCollection',
+		arguments: [this],
+		title: ''
+	};
+
+
+	async getDocuments(): Promise<any> {
+		let dbLink: string = this.db.getDbLink();
+		let client = new DocumentClient(this.db.server.getEndpoint(), {masterKey: this.db.server.getPrimaryMasterKey() });
+		let collSelfLink = dbLink + "/colls/" + this.id;
+		let docs = await this.readOneCollection(collSelfLink, client);
+		return await docs; 
+	}
+
+	async readOneCollection(selfLink, client): Promise<any>{
+		let documents = await client.readDocuments(selfLink, {maxItemCount : 20});
```

Comment:
Looks like the maxDocs setting got checked in after I started pushing this in. Give it a configurable setting after this check-in.

Reply:
Sounds good - thanks for creating an issue on it (I assume that's what this is for? https://github.com/Microsoft/vscode-cosmosdb/issues/88)